proposals/0159-relax-precompile-failure-constraint.md

```diff
@@ -0,0 +1,62 @@
+---
+simd: '0159'
+title: Relax Precompile Failure Constraint
+authors: Justin Starry (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-07-25
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+The Solana protocol currently prohibits block producers from including transactions
+with precompile instructions that fail verification. This constraint is not ideal
+since block producers have to do work to verify precompiles without a guarantee
+that they can receive a fee from the transaction.
+
+## Motivation
+
+Ensure that block producers are sufficiently incentivized to include transactions
+which make use of precompile instructions by allowing such transactions to be
+included in a block even if verification fails.
+
+## Alternatives Considered
+
+NA
```

Comment:
Have we considered to also relax the constraint similarly for all non-fee-payer signatures. i.e. transaction.signatures()[1..] in addition to any pre-compile signatures

Reply:
Yeah that's a good idea as well, I wouldn't mind expanding this SIMD to include those if that's what you're suggesting. It's a very similar constraint.

Reply:
Yeah think we make the relaxations part of the same SIMD.

Reply:
As discussed offline, we cannot relax the non-fee payer signature constraint because those signatures are not signed by the fee payer and could be tampered with in-flight.